sunpy/visualization/imageanimator.py

```diff
@@ -501,16 +501,10 @@ class ImageAnimator(BaseFuncAnimator):
     button_func: list
         List of functions to map to the buttons
 
-    Extra keywords are passed to imshow.
     """
 
     def __init__(self, data, image_axes=[-2, -1], axis_range=None, **kwargs):
 
-        self.naxis = data.ndim
-        self.num_sliders = self.naxis - 2
-        if len(image_axes) != 2:
-            raise ValueError("There can only be two spatial axes")
-
         all_axes = list(range(self.naxis))
```

Comment:
It's defined in ImageAnimator and LineAnimator before the init of ArrayAnimator is called.  I've moved this around so it's defined in ArrayAnimator.